### PR TITLE
Setup spa boilerplate

### LINKED ISSUE #12: creating React boilerplate
I was thinking we could pair up the 3 of us together to get you guys started with building the project, but this going to happen after we polish up the issues are do a bit more planning and more stuff, we are also going to cover the following in this boilerplate:

- [x] Webpack config
- [x] Babel config
- [x] ESLint config
- [x] basic React with styled components for CSS

and probably there is more

### PR DESCRIPTION
**Pull request by Karyum**
close #12

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    path: path.resolve(__dirname, 'dist'),
+    filename: 'bundle.js',
+    publicPath: '/',
+  },
+  devtool: 'inline-source-map',
+  devServer: { contentBase: './dist' },
+  module: {
+    loaders: [
+      {
+        test: /\.jsx?$/,
```
Comment (Amirk390) [inline webpack.config.js:16]: What does the `/\` means? :sweat_smile: 

didn't really found explanation about that in google :see_no_evil:

Comment (SamiSha99) [inline webpack.config.js:16]: Meaning of ```/\.jsx?$/,```?

**Review by SamiSha99 — APPROVED** [review]

File: src/App.jsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const Title = styled.h1`
```
Comment (Amirk390) [inline src/App.jsx:4]: :+1: 
thats going to be VERY useful, good thing to know

File: src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  color: blue;
+`;
```
Comment (Amirk390) [inline src/App.jsx:6]: Why you're using a different line?
is it just a cosmetic decision or will it break the code if you will use:
```
color: blue;`;
```

**Review by Amirk390 — APPROVED** [review]

File: index.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>Water Con</title>
```
Comment (matthewdking) [inline index.html]: Water Con? Is this the name of your app? What does water con mean? :man_shrugging:

**Review by matthewdking — COMMENTED** [review]
Just one question otherwise LGTM :taco:

File: index.html (hunk 4 above)
Comment (Amirk390) [reply index.html]: @matthewdking `Water-you-talking-about` :see_no_evil:

File: webpack.config.js (hunk 1 above)
Comment (Karyum) [reply webpack.config.js:16]: yeah umm this is just regex, it's also how Webpack takes file extension tests. so let's say you want to also compile CSS files you would do a different test like this `test: /\.css$/,`  which again is testing for file names the **end** with `.css`

Comment (Karyum) [reply webpack.config.js:16]: https://github.com/FACN3/Water-you-talking-about/pull/25#discussion_r166539866

in this case it's looking for filenames that end with `.jsx` or `.js` the `?` makes the letter before optional

File: src/App.jsx (hunk 3 above)
Comment (Karyum) [reply src/App.jsx:6]: it's just looks awful like this 
```
color: blue;`;
```

File: index.html (hunk 4 above)
Comment (Karyum) [reply index.html]: ohhhhh yeah let's change this

Comment (Karyum) [reply index.html]: i'll be guessing it's short for Consumption

Comment (Amirk390) [reply index.html]: @matthewdking 
> Water Con? Is this the name of your app? What does water con mean?

Fixed :+1: :smile:

**Comment by Amirk390** [comment]
@matthewdking 
> Water Con? Is this the name of your app? What does water con mean?

Fixed :+1: :smile:

**Review by matthewdking — APPROVED** [review]

**Pull request merged**